monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/HttpRequestStatusCounter.java

```diff
@@ -0,0 +1,45 @@
+package com.github.nhenneaux.resilienthttpclient.monitoredclientpool;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class HttpRequestStatusCounter {
+
+    private final AtomicInteger totalRequestCount;
```

Comment:
only count failure within `SingleIpHttpClient` directly?

Reply:
`totalRequestCount` would be needed for stats also. Nice indication if needing rate or logging